lib/ruby_lsp/listeners/hover.rb

```diff
@@ -180,6 +182,8 @@ def handle_method_hover(message, inherited_only: false)
 
       sig { params(name: String).void }
       def handle_instance_variable_hover(name)
+        return if @sorbet_level == Document::SorbetLevel::Strict
```

Comment:
On `typed: strict` and above, Sorbet enforces that all instance variables be declared and can therefore provide all features on them.